File: API/Services/Post/Post.Domain/Dtos/CommentPostDto.cs
namespace Post.Domain.Dtos
{
    public class CommentPostDto
    {
        public List<DisplayCommentDto> Comments { get; set; }
        public List<TreePathDto> TreePaths { get; set; }
    }

    public class TreeComment
    {
        public int Id { get; set; }
        public string Author { get; set; }
        public string Content { get; set; }
        public int? AncestorId { get; set; }
        public DateTimeOffset CreateDate { get; set; }
    }
}

File: Angular/projects/post-app/src/app/entites/comment.post.dto.ts
export interface TreeComment{
    id: number;
    author: string;
    content: string;    
    ancestorId: number | null;
    createDate: Date;
}

export interface DisplayCommentDto{
    id: number;
    author: string;
    content: string;
}

export interface CreateCommentDto{
    postId: string;
    author: string;
    content: string;
    ancestorId: number | null;
}